rfcs/0087-aarch64-tier1.md

```diff
@@ -0,0 +1,72 @@
+---
+feature: aarch64-tier1
+start-date: 2021-03-09
+author: Vika Shleina
+co-authors: TBD
+shepherd-team: TBD
+shepherd-leader: TBD
+related-issues: TBD
+---
+
+# Summary
+[summary]: #summary
+
+Move `aarch64-linux` from a Tier 2 platform to Tier 1, as described in [RFC 0046](/rfcs/0046-platform-support-tiers.md)
+
+# Motivation
+[motivation]: #motivation
+
+`aarch64-linux` support in Nixpkgs and NixOS matures over time and becomes
+more and more stable, and more devices appear having NixOS on ARM support.
+Moving it to a Tier 1 platform will allow us to block release channels on
+aarch64-related build failures, making it easier and safer for ARM users
+to upgrade their systems, and will help in keeping software versions in
+sync between several architectures due to `x86_64-linux` and `aarch64-linux`
+builds sharing a channel.
+
+`aarch64-linux` will benefit from increased binary cache coverage as an
+additional result, saving on build times for end users.
+
+# Detailed design
+[design]: #detailed-design
+
+If this RFC is accepted, `aarch64-linux` builds will be added to stable
+and unstable channels' `tested` aggregate jobs on Hydra, giving them ability
+to block channel advances. Hydra will start building aarch64 packages and run
+aarch64-based tests as part of stable and unstable channels, including them in
+the binary cache, increasing its coverage as a result.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+<!-- This section illustrates the detailed design. This section should clarify all
+confusion the reader has from the previous sections. It is especially important
+to counterbalance the desired terseness of the detailed design; if you feel
+your detailed design is rudely short, consider making this section longer
+instead. -->
+
+In [nixos/release-combined.nix](https://github.com/NixOS/nixpkgs/blob/master/nixos/release-combined.nix)
+`aarch64-linux` will be moved to `supportedSystems`, enabling NixOS tests
+to block channel advances in case of failures.
+
+(maybe something more? IDK for now)
```

Comment:
```suggestion
Merging this RFC should happen simultaneously with the merging of documentation and perhaps
a NixOS module for configuring qemu-binfmt as an aarch64 builder on x86_64 machines.

The list of NixOS AMIs on NixOS.org will also be extended to include aarch64 images.
```

Reply:
Also @grahamc may I add you as a co-author on the RFC? your contributions to the text seem substantial enough.

Reply:
You can add me, sure. Thanks for taking the initiative here.

> Please elaborate on the NixOS module part. We have boot.binfmt.emulatedSystems in configuration.nix - is this related?

Nice! I guess that part may be totally done, then :).

Reply:
At one time in the past, building with qemu-user+binfmt was iffy. It might have been specific to armv6 though. I personally wouldn't consider suggesting qemu-user+binfmt until it has been shown that a **full** rebuild without nixos.org cache works for aarch64.